client/extensions/woocommerce/state/sites/settings/stripe-connect-account/reducer.js

```diff
@@ -85,15 +90,52 @@ function connectAccountFetchComplete( state = {}, action ) {
 		error: action.error || '',
 		firstName: action.firstName || '',
 		isActivated: action.isActivated || false,
+		isDisconnecting: false,
 		isRequesting: false,
 		lastName: action.lastName || '',
 		logo: action.logo || '',
 	} );
 }
 
+/**
+ * Updates state to indicate account disconnection request is in progress
+ *
+ * @param  {Object} state  Current state
+ * @return {Object}        Updated state
+ */
+function connectAccountDisconnect( state = {} ) {
```

Comment:
I know that this is specific to the product ( stripe connect ) but connectAccountDisconnect sounds confusing. I don't know if there is any better name here - just an observation.

Reply:
Agreed. Renamed everything I could to deauthorize in https://github.com/Automattic/wp-calypso/pull/19244/commits/2d81a7a15a8b02918fd6b2ef39b9a54500e0b52c